lib/utils/cli.go

```diff
@@ -379,16 +411,20 @@ func needsQuoting(text string) bool {
 	return false
 }
 
+// defaultCommandPrintfWidth is the printf width of the command name with
+// padding when printing usage.
+const defaultCommandPrintfWidth = 12
+
 // Usage template with compactly formatted commands.
-var defaultUsageTemplate = `{{define "FormatCommand"}}\
+var defaultUsageTemplate = fmt.Sprintf(`{{define "FormatCommand"}}\
 {{if .FlagSummary}} {{.FlagSummary}}{{end}}\
 {{range .Args}} {{if not .Required}}[{{end}}<{{.Name}}>{{if .Value|IsCumulative}}...{{end}}{{if not .Required}}]{{end}}{{end}}\
 {{end}}\
 
 {{define "FormatCommands"}}\
 {{range .FlattenedCommands}}\
 {{if not .Hidden}}\
-  {{.FullCommand | printf "%-12s" }}{{if .Default}} (Default){{end}} {{ .Help }}
+  {{.FullCommand | printf "%%-%ds" }}{{if .Default}} (Default){{end}} {{ .Help }}
```

Comment:
Since we rely on templates, could we instead replace a chunk of the template with a function that returns the width? Ie, something like `defaultUsageTemplate = baseTemplate + defaultWidthFunc`, so we can swap the last part without having to find-and-replace strings in the text.

Reply:
the template is executed in kingpin source code, and the functions are defined there too. I am certainly ok adding some stuff there but not sure it will be harder to track if we have to update kingpin later.

Reply:
I always have to look up how to do templates, so this may not be correct, but I mean something like this:

```go
var defaultUsageTemplate = baseTemplate + defaultWidthTemplate

var baseTemplate = `
(...)

{{define "FormatCommands"}}\
{{range .FlattenedCommands}}\
{{if not .Hidden}}\
  {{.FullCommand | DefaultWidth }}{{if .Default}} (Default){{end}} {{ .Help }}
{{end}}\
{{end}}\
{{end}}\

(...)`

// Replace this chunk with our custom width.
var defaultWidthTemplate = `{{define "DefaultWidth"}}12s{{end}}`
```

Reply:
Gotcha, let me give that a try! Thanks

Reply:
i tried a few things, unfortunately none worked well. Turned out kingpin is using their custom fork of `text/template` of go 1.4 with the addition handling on `\`. If it's latest version of go template, i can actually do the entire loop i have here inside the template.

I've decided to make a quick patch in our kingpin to use latest  "text/template"

Reply:
Let me know if you'd rather keep the original change, it works for me.